[Huang Jing Yu] IP

jingyu987 opened the pull request:
## Fishron

> “Your mind is for having ideas, not holding them.” – David Allen (source)

Fishron helps your mind be free like a fish

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here.](https://www.youtube.com/watch?v=MtN1YnoL46Q&ab_channel=forrestfire101)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉


And it is **_FREE!_**

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)
##
If you Java programmer, you can use it to practice Java too. Here's the main method:


```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jingyu987 committed:
Level 0. Rename, Greet, Exit
(a) Give your chatbot a new name, to differentiate it from the placeholder name Duke.

(b) Implement an initial skeletal version of the it that simply greets the user and exits.
Example:

jingyu987 committed:
Level 1. Echo
Improve the skeletal version of Duke so that it echos commands entered by the user, and exits when the user types the command bye.

jingyu987 committed:
Level 2. Add, List
Add the ability to store whatever text entered by the user and display them back to the user when requested.

jingyu987 committed:
Level 3. Mark as Done
Add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

jingyu987 committed:
Level 5. Handle Errors
Teach the chatbot to deal with errors such as incorrect inputs entered by the user.

jingyu987 committed:
Level 4/5.
ToDos, Events, Deadlines
Add support for tracking three types of tasks:

ToDos: tasks without any date/time attached to it e.g., visit new theme park
Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
Events: tasks that start at a specific date/time and ends at a specific date/time
e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019

Handle Errors
Teach the chatbot to deal with errors such as incorrect inputs entered by the user.

jingyu987 committed:
Level 6. Delete
Add support for deleting tasks from the list.
Added more error handling as well.

jingyu987 committed:
Level 6. Delete
Add support for deleting tasks from the list.
Added more error handling as well.

jingyu987 committed:
A-TextUiTesting
↳ Test using the I/O redirection technique

Use the input/output redirection technique to semi-automate the testing of Duke.

jingyu987 committed:
A-TextUiTesting
↳ Test using the I/O redirection technique

Use the input/output redirection technique to semi-automate the testing of Duke.

jingyu987 committed:
Add Increments as branches: Level-7, Level-8
Do Level 7 in a branch named branch-Level-7. Here are the steps:
Start a branch branch-Level-7.
Implement Level 7 while committing to that branch at appropriate points,
Merge the branch back to the master branch (remember to create a merge commit i.e., no fast-forward).
Git tag the merge commit in the master branch as usual (i.e., add the tag Level-7).
Push the master branch, push the branch-Level-7, and push the tag, to your fork.
 Advanced git users: do not delete the branch after merging.
 Only merged branches are detected by the script. After merging a branch b1 to the master branch, you need to push both the master and the b1 branches to the fork. Pushing the master branch does not automatically take the b1 branch along with it just because it is already merged to the master branch.

jingyu987 committed:
Level 8. Dates and Times
Teach the chatbot how to understand dates and times. For example, if the command is deadline return book /by 2/12/2019 1800, the chatbot should understand 2/12/2019 1800 as 2nd of December 2019, 6pm, instead of treating it as just a String.

Minimal: Store deadline dates as a java.time.LocalDate (or java.time.LocalDateTime) in your task objects. Accept dates in a format such as yyyy-mm-dd format (e.g., 2019-10-15) and print in a different format such as MMM dd yyyy e.g., (Oct 15 2019).
Stretch goal: Use dates and times in more meaningful ways. e.g., add a command to print deadlines/events occurring on a specific date.

jingyu987 committed:
A-MoreOOP
↳ Make the code more OOP

Refactor the code to extract out closely related code as classes.

Minimal: Extract the following classes:
Ui: deals with interactions with the user
Storage: deals with loading tasks from the file and saving tasks in the file
Parser: deals with making sense of the user command
TaskList: contains the task list e.g., it has operations to add/delete tasks in the list

jingyu987 committed:
A-Packages
↳ Divide classes into packages

Organize the classes into suitable java packages.

Minimal: put all classes in one package e.g., duke
Stretch goal: divide into multiple packages as the number of classes increase e.g., duke.task, duke.command

jingyu987 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jingyu987 committed:
Merge commit '2ae1fb4347f2554ad53b732b414a31a4c5ec322f'

# Conflicts:
#	src/main/java/Fishron.java
#	text-ui-test/EXPECTED.TXT

jingyu987 committed:
A-JUnit
↳ Add JUnit tests

Add JUnit tests to test the behavior of the code.

Minimal: Use JUnit to at least two non-trivial methods, from two different classes (if you have multiple classes),
and, ensure they are tested reasonably well (i.e., the test code should try to catch most potential bugs in the target methods).
Stretch goal: Use JUnit to test all non-trivial public methods of all classes.

jingyu987 committed:
Add JavaDoc comments

Add JavaDoc comments to the code.

Minimal: Add header comments to at least half of the non-private classes/methods.
Stretch goal: Add header comments to all non-private classes/methods, and non-trivial private methods.

jingyu987 committed:
Add JavaDoc comments

Add JavaDoc comments to the code.

Minimal: Add header comments to at least half of the non-private classes/methods.
Stretch goal: Add header comments to all non-private classes/methods, and non-trivial private methods.

jingyu987 committed:
Merge branch 'branch-A-JavaDoc'

jingyu987 committed:
A-CodingStandard
↳ Tweak the code to comply with a coding standard

Tweak the code to comply with a given coding standard. From this point onward, ensure any new code added are compliant with the given coding standard.

jingyu987 committed:
Merge branch 'branch-A-CodingStandard'

jingyu987 committed:
Level 9. Find
Give users a way to find a task by searching for a keyword.

jingyu987 committed:
Merge branch 'branch-Level-9'

m1oojv commented on src/main/java/fishron/Command.java:
```diff
@@ -0,0 +1,41 @@
+     * @param storage  The storage to read from or write to.
+     */
+    public void execute(TaskList taskList, Ui ui, Storage storage)  {
+        // This method is overridden by concrete command classes.
```
Good Explanation Comment. Very clear and helps in understanding.

m1oojv commented on src/main/java/fishron/Parser.java:
```diff
@@ -0,0 +1,180 @@
+            String[] parts = input.split("todo");
+            return new AddCommand(parseTodo(parts[1]));
+
+        } else if (input.toLowerCase().startsWith("deadline")) {
```
good use of toLowerCase() as it helps reduce user incorrect input ! :D

m1oojv commented on src/main/java/fishron/Parser.java:
```diff
@@ -0,0 +1,180 @@
+            throw new FishronException("☹ OOPS!!! Please provide a task to find.");
+        }
+
+        if (command.toLowerCase().startsWith("mark") && command.split("mark").length <= 1) {
```
might want to consider using `switch` instead of `if....else...`

m1oojv reviewed (commented):
Overall, I found your code easy to read for the most part. Really good OOP and documentation. Just made one slight optional suggestion for improvement.

billieboy7 commented on src/main/java/fishron/AddCommand.java:
```diff
@@ -0,0 +1,35 @@
+package fishron;
```
You can consider creating a separate package for the Command class and its subclasses, instead of packing everything under the "fishron" package. Will be helpful as the project gets bigger :)

Chen-Kuei replied:
yes, since you created a lot of classes, it may be a bit difficult for the readers to understand the structure of your program.

billieboy7 commented on src/main/java/fishron/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+public class Deadline extends Task {
+    private LocalDateTime by;
+
+    /**
```
I like your javadoc explanations. They are clear and succinct. Great job!

billieboy7 commented on src/main/java/fishron/Storage.java:
```diff
@@ -0,0 +1,92 @@
+        } catch (FileNotFoundException e) {
+            System.out.println("No saved tasks found.");
+        } catch (FishronException e) {
+            System.out.println("____________________________________________________________");
```
You can consider adding these dividers as a separate method in the UI class. Makes it easier and standardised.

billieboy7 commented on src/main/java/fishron/Parser.java:
```diff
@@ -0,0 +1,180 @@
+
+        } else if (input.toLowerCase().startsWith("event")) {
+            String[] parts = input.split("/from|/to");
+            return new AddCommand(parseEvent(parts[0].split("event")[1].trim(), parts[1].trim(), parts[2].trim()));
```
Good use of trim() here. Avoids any user-generated error caused by additional spaces.

jingyu987 committed:
Update README.md

Signed-off-by: Jing Yu <123240462+jingyu987@users.noreply.github.com>

jingyu987 committed:
Update README.md

Signed-off-by: Jing Yu <123240462+jingyu987@users.noreply.github.com>

jingyu987 committed:
Add checkStyle to detect coding style violations.

jingyu987 committed:
Merge branch 'branch-A-CheckStyle'

jingyu987 committed:
Add GUI and proper packaging

jingyu987 committed:
Merge branch 'branch-level-10'

Chen-Kuei commented on src/main/java/Fishron.java:
```diff
@@ -2,6 +2,8 @@
 
 public class Fishron {
     private String name;
+    private String[] storage = new String[100];
```
can consider creating a **'List' class** here to make your program more OOP!

Chen-Kuei reviewed (commented):
Good job! 
Just need to have some changes to make your program more well-organised.

jingyu987 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	README.md

jingyu987 committed:
Update README.md

Signed-off-by: Jing Yu <123240462+jingyu987@users.noreply.github.com>

jingyu987 committed:
Update README.md

Signed-off-by: Jing Yu <123240462+jingyu987@users.noreply.github.com>

jingyu987 committed:
Add assertions.

Added assertions in some key places.

Ensures that important assumptions hold at various points in the code.

jingyu987 committed:
Add assertions

Added assertions in some key places.

Ensures that important assumptions hold at various points in the code.

jingyu987 committed:
Merge branch 'branch-A-Assertions' of https://github.com/jingyu987/ip into branch-A-Assertions

jingyu987 committed:
Merge pull request #1 from jingyu987/branch-A-Assertions

Branch a assertions

jingyu987 committed:
Update readme

jingyu987 committed:
Merge branch 'master' of https://github.com/jingyu987/ip

jingyu987 committed:
Improve code quality

Improved the general code quality of the project.

jingyu987 committed:
Merge pull request #2 from jingyu987/branch-A-CodeQuality

Improve code quality

jingyu987 committed:
Add help command

A user currently does not have a way to view the functions
available and how to use them.

A user will be unsure how to use the program, not knowing the
commands available.

Lets add a "help" command that allows users to view the commands
available to them.

Using a help command is a simple way for users to easily
understand how to use the program.

jingyu987 committed:
Improve code quality and add user guide

jingyu987 committed:
Improve code quality and fix storage path
